test/kldivergences.jl

```diff
@@ -0,0 +1,86 @@
+@testset "KL divergences" begin
+    function test_kl(p, q)
+        @test kldivergence(p, q) > 0
+        @test kldivergence(p, p) ≈ 0 atol=1e-1
+        @test kldivergence(q, q) ≈ 0 atol=1e-1
+        if p isa UnivariateDistribution
+            @test kldivergence(p, q) ≈ invoke(kldivergence, Tuple{UnivariateDistribution,UnivariateDistribution}, p, q) atol=1e-1
+        elseif p isa MultivariateDistribution
+            @test kldivergence(p, q) ≈ invoke(kldivergence, Tuple{MultivariateDistribution,MultivariateDistribution}, p, q; nsamples=10000) atol=1e-1
+        end
+    end
+    
+    @testset "Categorical" begin
+        @test kldivergence(Categorical([0.0, 0.1, 0.9]), Categorical([0.1, 0.1, 0.8])) ≥ 0
+        @test kldivergence(Categorical([0.0, 0.1, 0.9]), Categorical([0.1, 0.1, 0.8])) ≈
+            kldivergence([0.0, 0.1, 0.9], [0.1, 0.1, 0.8])
+        @test expectation(Normal(0.0, 1.0), identity, 1e-10) ≤ 1e-9
```

Comment:
Doesn't really fit in this test set :smile: It's also one of the deprecated calls (which might be good to test as well).

What about:
- moving the `kldivergence` of ´`Categorical` to the univariate section below?
- have a separate section before the KL divergence test sets for expectations?

I think we should compute expectations of e.g. a normal distribution, a Poisson distribution (unbounded discrete) and a Binomial distribution (bounded discrete), with the new and the deprecated calls (the deprecation could also be checked with `@test_deprecated`).

The file name is not completely correct then anymore, so maybe we could just rename it to `functionals.jl` again (which is also consistent with the source file src/functionals.jl).